Casks/o/ogdesign-eagle.rb

```diff
@@ -1,19 +1,19 @@
 cask "ogdesign-eagle" do
-  arch arm: "M1-"
+  arch arm: "arm64", intel: "x64"
 
-  version "3.0,44"
-  sha256 arm:   "0388a36a422b3793771cac024824af97fb4af5f646ba2f2c269a04006cc5ddb7",
-         intel: "a990231ebf1554d9029531a91868ec6f361c4fd917d30479e0e9ffd17062db5b"
+  version "4.0,1"
+  sha256 arm:   "f2f75d2885b88f6328afd7e21898b17070b33e4c738547443851a09c7c4eb01d",
+         intel: "5b0cdce85505359fd40b58f2bb77c47cc67f6b4931fc24fa433cf870c162c0bd"
 
-  url "https://eaglefile.oss-cn-shenzhen.aliyuncs.com/releases/Eagle-#{version.csv.first}-#{arch}build#{version.csv.second}.dmg",
+  url "https://eaglefile.oss-cn-shenzhen.aliyuncs.com/releases/Eagle-#{version.csv.first}-#{arch}-build#{version.csv.second}.dmg",
       verified: "eaglefile.oss-cn-shenzhen.aliyuncs.com/releases/"
   name "Eagle"
   desc "Organise all your reference images in one place"
   homepage "https://eagle.cool/"
 
   livecheck do
     url "https://eagle.cool/check-for-update"
-    regex(/Eagle[._-]v?(\d+(?:\.\d+)+)-#{arch}build(\d+(?:\.\d+)*)\.dmg/i)
+    regex(/Eagle[._-]v?(\d+(?:\.\d+)+)-#{arch}-build(\d+(?:\.\d+)*)\.dmg/i)
```

Comment:
```suggestion
    regex(/Eagle[._-]v?(\d+(?:\.\d+)+)-#{arch}[._-]?build(\d+(?:\.\d+)*)\.dmg/i)
```

Looking back, upstream has used `build`, `_build`, and `-build` at various points, so we should account for all of those situations.